src/test/ui/qualified/qualified-path-params.stderr

```diff
@@ -1,4 +1,4 @@
-error[E0533]: expected unit struct/variant or constant, found method `<<S as Tr>::A>::f<u8>`
+error[E0533]: expected unit struct/variant or constant, found method `<S as Tr>::A::f<u8>`
```

Comment:
This change is not _incorrect_, but it _feels_ like a mild regression.

Reply:
Hmm, it's probably fine but how does it compose with type arguments and turbofish? Also why are we using HIR pretty printing in diagnostics?

Reply:
It's supposed to be only a fallback for when we can't compute a snippet, but that shouldn't really happen in practice. Some older codepaths are using `pprint` but we should move them all to prefer `span_to_snippet`.

Reply:
Can you make a ticket for that?

Reply:
https://github.com/rust-lang/rust/issues/65568

Reply:
I could try to improve it, so that additional parentheses are added for nesting > 1 - that way we'd have e.g. `Foo::new()` and `<Foo::bar>::new()`. Do you think it's worth changing?

Reply:
Sorry for the silence, had missed your response @Patryk27.

> Do you think it's worth changing?

If you feel you can do it, I think that would be ideal.

Reply:
Fixed :-)